### PR TITLE
Update Azure SQL server guide

### PR DESCRIPTION
**Pull request by gabrielcorado**
This PR updates the Azure SQL server guide with the following:
* Change the Azure service principal to the system-assigned identity [to prevent an SDK initialization error](https://github.com/gravitational/teleport/issues/20513).
* Add missing permission to read virtual machine identities.
* Add a troubleshooting scenario where the Teleport Database service cannot reach the database.

### CODE REVIEW

**Review by GavinFrazar — APPROVED** [review]
LGTM

**Comment by alexfornuto** [comment]
Can we use the https://github.com/gravitational/teleport/blob/master/docs/pages/includes/database-access/azure-configure-service-principal.mdx partial file in this doc to reduce duplicate content?

**Comment by gabrielcorado** [comment]
> Can we use the https://github.com/gravitational/teleport/blob/master/docs/pages/includes/database-access/azure-configure-service-principal.mdx partial file in this doc to reduce duplicate content?

@alexfornuto Although users can still use user-managed identities and applications to set Teleport permissions, we need them to enable system-managed identities. Those two steps might need to be clarified as they overlap in giving Teleport a service principal. This PR changes that so that we have a single stage where users:
1. Enable the system-managed identity on their VM.
2. Use this identity to set up Teleport permissions.

I agree there are some duplicate contents on both pages, we could add the system-managed identity into the partial file, but we would require it to be the first tab, which I wonder would benefit all other places.

**Comment by GavinFrazar** [comment]
my 2 cents: Keep it separate, since SQL Server is a special case where we need them to use a system-assigned identity. Other protocols can use a user-managed identity

**Comment by alexfornuto** [comment]
Thanks for the detailed responses!

File: docs/pages/database-access/guides/azure-sql-server-ad.mdx [hunk 1]
```diff
@@ -62,7 +62,88 @@ login to SQL Server.
...
+
+![Create JSON role](../../../img/database-access/guides/sqlserver/create-role-from-json.png)
+
+## Step 5/9. Configure virtual machine identities
+
+On the **Teleport Database Service virtual machine instance** **Identity**
+section, enable the system assigned identity. This is used by Teleport to access
+Azure APIs.
+
+![System assigned identity page](../../../img/database-access/guides/sqlserver/system-managed-identity.png)
```
Comment (alexfornuto) [inline docs/pages/database-access/guides/azure-sql-server-ad.mdx:132]: Could/should the screenshot be expanded to show the nav bar and/or breadcrumbs, to help the user find/identify this section of the instance settings? Similar to the screenshot two above.

**Review by alexfornuto — APPROVED** [review]
I made some edits, commit-separated by type so you can play with them as you like. Generally 👍, with one outstanding nit.

**Comment by alexfornuto** [comment]
Oopsie, this didn't get any backport labels 🤷

**Comment by gabrielcorado** [comment]
@alexfornuto I'll backport it to v11 too, but it will be slightly different from this one as the v11 version doesn't have auto-discovery.

**Pull request merged**